server/channels/app/content_flagging.go

```diff
@@ -396,3 +396,261 @@ func (a *App) GetPostContentFlaggingPropertyValues(postId string) ([]*model.Prop
 
 	return propertyValues, nil
 }
+
+func (a *App) PermanentDeleteFlaggedPost(c request.CTX, actionRequest *model.FlagContentActionRequest, reviewerId string, flaggedPost *model.Post) *model.AppError {
+	// when a flagged post is removed, the following things need to be done
+	// 1. Hard delete corresponding file infos
+	// 2. Hard delete file infos associated to post's edit history
+	// 3. Hard delete post's edit history
+	// 4. Hard delete the files from file storage
+	// 5. Hard delete post's priority data
+	// 6. Hard delete post's post acknowledgements
+	// 7. Hard delete post reminders
+	// 8. Scrub the post's content - message, props
+
+	status, appErr := a.GetPostContentFlaggingStatusValue(flaggedPost.Id)
+	if appErr != nil {
+		return appErr
+	}
+
+	if status == nil {
+		return model.NewAppError("removeFlaggedPost", "api.content_flagging.error.post_not_flagged", nil, "", http.StatusBadRequest)
+	}
+
+	statusValue := strings.Trim(string(status.Value), `"`)
+	if statusValue != model.ContentFlaggingStatusPending && statusValue != model.ContentFlaggingStatusAssigned {
+		return model.NewAppError("removeFlaggedPost", "api.content_flagging.error.post_not_in_progress", nil, "", http.StatusBadRequest)
+	}
+
+	editHistories, appErr := a.GetEditHistoryForPost(flaggedPost.Id)
+	if appErr != nil {
+		editHistories = []*model.Post{}
+
+		if appErr.StatusCode != http.StatusNotFound {
+			c.Logger().Error("PermanentlyRemoveFlaggedPost: Failed to get edit history for flaggedPost", mlog.Err(appErr), mlog.String("post_id", flaggedPost.Id))
+		}
+	}
+
+	for _, editHistory := range editHistories {
+		if filesDeleteAppErr := a.PermanentDeleteFilesByPost(c, editHistory.Id); filesDeleteAppErr != nil {
```

Comment:
question: are this expected to fail "silently" only logging the errors instead of returning them?